### PR TITLE
Add trace context fields to GcpLayout.json

### PR DESCRIPTION
**Pull request by aabmass**
This PR adds a few tracing related JSON fields to the included `GcpLayout.json` event template which are documented in the [Cloud Logging docs](https://cloud.google.com/logging/docs/structured-logging#structured_logging_special_fields).

The intended source is [well known OpenTelemetry MDC keys](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/v2.3.0/docs/logger-mdc-instrumentation.md) encoded as a hex strings following the [W3C Trace Context standard](https://www.w3.org/TR/trace-context/#trace-id).

- MDC `trace_id` -> `logging.googleapis.com/trace`
- MDC `span_id` -> `logging.googleapis.com/spanId`
- `true` -> `logging.googleapis.com/trace_sampled`. This is unconditionally set to true because there is currently no way to derive it from W3C `trace_flags` with JTL (see https://github.com/apache/logging-log4j2/discussions/2482). I'm OK to remove this if needed, as it will affect existing users even if the MDC keys are not set. However it's absence does break some GCP UI features.

### CODE REVIEW

**Review by vy — APPROVED** [review]
@aabmass, thanks so much for the contribution! Much appreciated! :bow:

A quick question just before merging it: If `trace_id` and/or `span_id` MDC values are missing, `logging.googleapis.com/trace_sampled` will still be set to `true` without `logging.googleapis.com/trace` and/or `logging.googleapis.com/spanId` fields. Will this still be okay?

I would also really appreciate it if you can create a `src/changelog/.2.x.x/add_GcpLayout_tracing_support.xml` changelog entry file.

**Comment by aabmass** [comment]
> A quick question just before merging it: If `trace_id` and/or `span_id` MDC values are missing, `logging.googleapis.com/trace_sampled` will still be set to `true` without `logging.googleapis.com/trace` and/or `logging.googleapis.com/spanId` fields. Will this still be okay?

Would appreciate a quick LGTM from @dashpole, @punya, or @jsuereth on this point before merging

**Review by jsuereth — APPROVED** [review]

**Comment by aabmass** [comment]
> I would also really appreciate it if you can create a `src/changelog/.2.x.x/add_GcpLayout_tracing_support.xml` changelog entry file.

Added

File: src/changelog/.2.x.x/add_GcpLayout_tracing_support.xml
Comment (vy) [inline src/changelog/.2.x.x/add_GcpLayout_tracing_support.xml:7]:
```suggestion
       type="added">
  <issue id="2498" link="https://github.com/apache/logging-log4j2/pull/2498"/>
  <description format="asciidoc">Add trace context fields to `GcpLayout.json`</description>
```

**Comment by vy** [comment]
@aabmass, thanks so much! I will merge this once CI passes and `cherry-pick` it to `main` too.

**Pull request merged**
